packages/emails/src/components/BaseEmailHtml.tsx

```diff
@@ -24,78 +25,137 @@ export const BaseEmailHtml = (props: {
   subtitle?: React.ReactNode | string;
   headerType?: BodyHeadType;
   hideLogo?: boolean;
+  locale: string;
 }) => {
+  const textDirection = getTextDirection(props.locale);
+  const isRTL = textDirection === "rtl";
```

Comment:
You can simply get the text direction by passing it to new `Intl.Locale(props.locale)` and extracting it using the `getTextInfo()` method

Reply:
hi, i did implement this

is a try catch block (because of browsers that dont support the Intl library) good enough?

 try {
    if (typeof Intl.Locale !== "undefined" && "getTextInfo" in Intl.Locale.prototype) {
      const extendedLocale = new Intl.Locale(locale) as ExtendedLocale;
      if (extendedLocale.getTextInfo) {
        return extendedLocale.getTextInfo().direction;
      }
    }
    // Fallback for browsers that don't support Intl.Locale.prototype.getTextInfo
    const rtlLocales = ["ar", "he", "fa", "ur"];
    return rtlLocales.some((lang) => locale.toLowerCase().startsWith(lang)) ? "rtl" : "ltr";
  } catch (error) {
    console.error(`Error determining text direction for locale ${locale}:`, error);
    return "ltr";
  }
  
  
  lmk if this looks good, have to fix some type errors in this